rust/kernel/spi.rs

```diff
@@ -0,0 +1,279 @@
+// SPDX-License-Identifier: GPL-2.0
+
+//! This module provides safer and higher level abstraction over the kernel's SPI types
+//! and functions.
+//!
+//! C header: [`include/linux/spi/spi.h`](../../../../include/linux/spi/spi.h)
+
+use crate::bindings;
+use crate::c_types;
+use crate::error::{Error, Result};
+use crate::str::CStr;
+use alloc::boxed::Box;
+use core::pin::Pin;
+
+/// Wrapper struct around the kernel's `spi_device`
+#[derive(Clone, Copy)]
+pub struct SpiDevice(*mut bindings::spi_device);
+
+impl SpiDevice {
+    /// Create an SpiDevice from a mutable spi_device raw pointer. This function is unsafe
+    /// as the pointer might be invalid.
+    ///
+    /// # Safety
+    ///
+    /// The pointer must be valid. This can be achieved by calling `to_ptr` on a previously
+    /// constructed, safe SpiDevice instance, or by making sure that the pointer points
+    /// to valid memory.
+    /// You probably do not want to use this abstraction directly. It is mainly used
```

Comment:
Paragraphs should be split by an empty line:

```suggestion
    /// to valid memory.
    /// You probably do not want to use this abstraction directly. It is mainly used
```

Reply:
Fixed by 92a19c2ef2d0d0780c34327b273a21db627af711